lifelines/utils/ShiftedBetaGeometric.py

```diff
@@ -0,0 +1,726 @@
+        # trivial model (with zero weights) approximates the mean of the
+        # target values. Here, the absence of weights (including bias) does
+        # NOT lead to a trivial model, but one with a unreasonable
+        # preference for alpha = exp(0) and beta = exp(0). ***
+        # Moreover different regularization parameters for alpha and beta
+        # can be helpful, specially in extreme cases when the distribution
+        # is near the extremes (0 or 1)
+        l2_reg = self.gammaa * sum(wa[1:]**2) + self.gammab * sum(wb[1:]**2)
+
+        # update log-likelihood with regularization val.
+        log_like -= l2_reg
+
+        # Get the full alpha and beta for each sample and store the values in
+        # ndarrays of shape (n_samples, )
+        alpha, beta = self._compute_alpha_beta(X, wa, wb)
+
+        # loop over data and add the contribution to the log-likelihood from
+        # each sample in the dataset.
+        # Notice that once we have alpha and beta for each sample we do not
+        # need the feature matrix to compute the contribution to the
+        # log-likelihood.
+        retention_probs = self._log_retention_stats(alpha, beta, age)
+
+        # Contribution to log likelihood based on status (alive vs. dead)
+        log_like += np.sum(retention_probs[0][np.where(alive == 0)[0]])
+        log_like += np.sum(retention_probs[1][np.where(alive == 1)[0]])
+
+        # Negative log_like since we will use scipy's minimize object.
+        return -log_like
+
+    def fit(self, X, age, alive, restarts=1):
+        """
+        Method responsible for the learning step it takes all the relevant data
+        as argument as well as the number of restarts with random seeds to
+        perform. While restarting with other seeds sounds like a good idea the
+        model has proven to be fairly stable and this may be removed in the
+        future.
+
+        :param X: ndarray of shape (n_samples, n_features)
+            The feature matrix
+
+        :param age: ndarray of shape (n_samples, )
+            An array with the age of each individual.
+
+        :param alive: ndarray of shape (n_samples, )
+            An array with
+
+        :param restarts: int
+            Number of times to run the optimization procedure with different seeds
+        """
+
+        # Make sure ages are all non negative...
+        min_age = min(age)
+        if min(age) < 1:
+            raise ValueError("All values of age must be equal or greater to "
+                             "one. The minimum value of "
+                             "{} was found.".format(min_age))
+
+        # Make sure alive is either zero or one!
+        alive_vals = set(alive)
+        if alive_vals != {0, 1}:
+            raise ValueError('Values for alive must be either zero or one. A '
+                             'value of '
+                             '{} was found.'.format(list(alive_vals - {0, 1})))
```

Comment:
My concern is that negative values will eventually raise an error when computing the beta function. Unless you know how the model work, such an error may look quite cryptic.

Perhaps I can cast alive to integer to handle both bool / float entries at once. This variable is lot more self explanatory, and erros from inputing improper values should be more self evident. You suggest simply not checking?